File: python/Euler_092.py
def number_chain_result(n):
    current = n
    while True:
        next = sum(int(i) ** 2 for i in str(current))
        if next == 1:
            return 1
        elif next == 89:
            return 89
        else:
            current = next


starting_nums = sum(1 for i in range(1, 10000000) if number_chain_result(i) == 89)
print(starting_nums)

File: java/Euler_092.java

public class Euler_092 {

    public static int numberChainResult(int n) {
        int current = n;
        while (true) {
            int next = 0;
            String[] digits = Integer.toString(current).split("");
            for (String digit : digits) {
                next += Math.pow(Integer.parseInt(digit), 2);
            }
            if (next == 1) {
                return 1;
            } else if (next == 89) {
                return 89;
            } else {
                current = next;
            }
        }
    }

    public static void main(String[] args) {
        int startingNums = 0;
        for (int i = 1; i < 10000000; i++) {
            if (numberChainResult(i) == 89) {
                startingNums++;
            }
        }
        System.out.println(startingNums);
    }

}
